frontend/src/metabase/questions/containers/QuestionIndex.jsx

```diff
@@ -49,81 +85,56 @@ export default class QuestionIndex extends Component {
-                    >
-                        <div className="flex align-center">
-                            <DisclosureTriangle open={questionsExpanded} />
-                            <h2>Everything Else</h2>
-                        </div>
-                    </div>
-                }
-                <Collapse isOpened={showQuestions && (questionsExpanded || !showCollections)} keepCollapsedContent={true}>
+            <div className={cx("relative mx4", {"full-height flex flex-column": showNoSavedQuestionsState})}>
+                { showNoCollectionsState && <CollectionEmptyState /> }
+
+                <QuestionIndexHeader collections={collections} isAdmin={isAdmin} onSearch={this.props.search} />
+
+                { hasCollections && <CollectionButtons collections={collections} isAdmin={isAdmin} push={push} /> }
+
+                { showNoSavedQuestionsState && <NoSavedQuestionsState /> }
+
+                { showEverythingElseTitle && <h2 className="mt2 mb2">Everything Else</h2> }
+
+                <div className={cx({ "hidden": !showEverythingElseContents })}>
+                    {/* EntityList loads `questions` according to the query specified in the url query string */}
                     <EntityList
                         entityType="cards"
-                        entityQuery={{ f: "all", collection: "", ...location.query }}
+                        entityQuery={{f: "all", collection: "", ...location.query}}
                         // use replace when changing sections so back button still takes you back to collections page
                         onChangeSection={(section) => replace({
                             ...location,
-                            query: { ...location.query, f: section }
+                            query: {...location.query, f: section}
                         })}
-                        defaultEmptyState="Questions that aren’t in a collection will be shown here"
                     />
-                </Collapse>
+                </div>
             </div>
         )
     }
 }
 
-const CollectionEmptyState = () =>
-    <div className="flex align-center p2 bordered border-med border-brand rounded bg-grey-0 text-brand">
-        <Icon name="collection" size={32} className="mr2"/>
-        <div className="flex-full">
-            <h3>Create collections for your saved questions</h3>
-            <div className="mt1">
-                Collections help you organize your questions and allow you to decide who gets to see what.
-                {" "}
-                <a href="http://www.metabase.com/docs/latest/administration-guide/06-collections.html" target="_blank">
-                    Learn more
-                </a>
-            </div>
-        </div>
-        <Link to="/collections/create">
-            <Button primary>Create a collection</Button>
-        </Link>
-    </div>
+const mapStateToProps = (state, props) => ({
```

Comment:
In most of the app we keep the `mapDispatchToProps` and `mapDispatchToProps` at the top. I think it's nice to see a component is `@connect`ed up front. Is there a reason you moved it?